### DISCUSSION TITLE
If there is a Distributed DML feature

### DISCUSSION THREAD

**Discussion by realcbb** [ClickHouse/ClickHouse#51705] [Q&A] [answered]
If there is a Distributed DML feature, it is much useful when same dml needs to be executed on every shard's local table. Can we have this feature?

**Comment by UnamedRus** [comment]
You can run 
https://clickhouse.com/docs/en/sql-reference/statements/alter/update

```
ALTER TABLE xxx ON CLUSTER 'xxxx' UPDATE xx=xxx WHERE xx
ALTER TABLE xxx ON CLUSTER 'xxxx' DELETE WHERE xx
```

**Reply by realcbb** [reply]
Sometimes needs to `insert into local_table_A select ... from local_table_B left join local_table_C on ... where ...`
local_table_A, local_table_B, local_table_C have the same sharding key.
In this case Distributed DML will has higher performance than deal with distributed tables.

**Reply by UnamedRus** [reply, marked as answer]
In that case use distributed table + │ parallel_distributed_insert_select                   │ 0        │       0 │ Process distributed INSERT SELECT query in the same cluster on local tables on every shard; if set to 1 - SELECT is executed on each shard; if set to 2 - SELECT and INSERT are executed on each shard                                                     │ ᴺᵁᴸᴸ │ ᴺᵁᴸᴸ │        0 │ UInt64

**Reply by realcbb** [reply]
Excellent! Thanks!
